### PR TITLE
CreatePair API implementation with Success tests

### PR DESCRIPTION
**Pull request by sharma-tapas**
CreatePair API implementation with CreatePair Success tests
Signed-off-by: Tapas Sharma <tapas@portworx.com>

**What this PR does / why we need it**:
This is the createpair implmentation for the SDK.

**Which issue(s) this PR fixes** (optional)
Part Of #653

### CODE REVIEW

**Comment by lpabon** [comment]
Travis failed because one of the checks noticed that the latest generated files have not been applied to the PR.

File: api/api.proto [hunk 1]
```diff
@@ -937,6 +937,28 @@ service OpenStorageCluster {
     }
 }
 
+// OpenStorageClusterPair service provides the methods to manage a cluster pair
+service OpenStorageClusterPair{
+  // Creates Pair with a remote cluster and returns details about the remote cluster
+  //
+  // ##### Example
+  // {% codetabs name="Golang", type="go" -%}
+  // id, err := client.Create(context.Background(), &api.SdkVolumeCreateRequest{
```
Comment (lpabon) [inline api/api.proto]: The comment is using SdkVolumeCreateRequest, it should be SdkClusterPairCreateRequest.

File: api/api.proto [hunk 2]
```diff
@@ -937,6 +937,28 @@ service OpenStorageCluster {
...
+  //   RemoteClusterIp: "127.0.0.1",
+  //   RemoteClusterPort: 12345,
+  //   RemoteClusterToken: "<Auth-Token>",
+  //   SetDefault: <Boolean-Value>,
+  // })
+  // {%- endcodetabs %}
+  rpc Create(ClusterPairCreateRequest)
```
Comment (lpabon) [inline api/api.proto]: [See the requirements](https://github.com/libopenstorage/openstorage/blob/master/docs/dev-sdk.md#sdk-requirements) All Sdk messages (arguments) should start with `SdkXXX`

Comment (lpabon) [reply api/api.proto]: So it looks like you probably should have something like this due to the current already defined messages:

```
message SdkClusterPairCreateRequest {
    ClusterPairCreateRequest req = 1;
}
```

File: api/api.proto [hunk 3]
```diff
@@ -937,6 +937,28 @@ service OpenStorageCluster {
...
+    returns (ClusterPairCreateResponse){
+      option(google.api.http) = {
+        post: "/v1/clusters/pair"
```
Comment (lpabon) [inline api/api.proto]: Just have "/v1/clusterspairs" since this is the ClusterPairs service

File: api/server/sdk/cluster_pair.go [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+	} else if len(req.GetRemoteClusterToken()) == 0 {
+		return nil, status.Error(codes.InvalidArgument, "Must supply Authentication Token")
+	} else if req.GetRemoteClusterPort() == 0 {
+		return nil, status.Error(codes.InvalidArgument, "Must supply Remote cluster Port")
+	}
+
+	resp, err := s.cluster.CreatePair(req)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Cannot create cluster with remote pair %v : %v",
+			req.GetRemoteClusterIp(), err.Error())
```
Comment (lpabon) [inline api/server/sdk/cluster_pair.go]: This should be `... pair %s: %v", req.GetRemoteClusterIp(), err)`

File: api/server/sdk/cluster_pair_test.go [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+		SetDefault:         false,
+	}
+	resp := &api.ClusterPairCreateResponse{
+		RemoteClusterId:   remoteClusterID,
+		RemoteClusterName: remoteClusterName,
+	}
+
+	s.MockCluster().
+		EXPECT().
+		CreatePair(&api.ClusterPairCreateRequest{
```
Comment (lpabon) [inline api/server/sdk/cluster_pair_test.go]: nice!

File: api/server/sdk/server.go [hunk 6]
```diff
@@ -147,6 +151,7 @@ func (s *Server) Start() error {
 		api.RegisterOpenStorageIdentityServer(grpcServer, s.identityServer)
 		api.RegisterOpenStorageMountAttachServer(grpcServer, s.volumeServer)
 		api.RegisterOpenStorageAlertsServer(grpcServer, s.alertsServer)
+		api.RegisterOpenStorageClusterPairServer(grpcServer, s.clusterPairServer)
 	})
```
Comment (lpabon) [inline api/server/sdk/server.go]: You need to add the new service to the REST gateway. https://github.com/libopenstorage/openstorage/blob/master/api/server/sdk/server.go#L293

**Comment by lpabon** [comment]
This needs a rebase to incorporate the updates to the ClusterPairResponse messages added in #662.

File: api/server/sdk/server.go (hunk 6 above)
Comment (sharma-tapas) [reply api/server/sdk/server.go]: Added the following
RegisterOpenStorageClusterPairHandlerFromEndpoint(

File: api/api.proto (hunk 2 above)
Comment (sharma-tapas) [reply api/api.proto]: I tried changing this to the SdkClusterPairRequest and Response, but that requires change in the interface, clients, and cluster.go files, to make everything adhere to this parameter in the create call.

**Pull request closed without merging**